ring/ring.go

```diff
@@ -188,9 +270,9 @@ func (r *Ring) Type() Type {
 	}
 }
 
-// setParameters initializes a *Ring by setting the required pre-computed values (except for the NTT-related values, which are set by the
-// genNTTParams function).
-func (r *Ring) setParameters(N int, Modulus []uint64) error {
+// SetParameters initializes a *Ring by setting the required pre-computed values (except for the NTT-related values, which are set by the
+// GenNTTParams function).
+func (r *Ring) SetParameters(N int, Modulus []uint64) error {
```

Comment:
This function reallocate everything so is semantically closer to a `New*` type of function. 

I don't think there is a case for making it public and it can probably be integrated directly into the `NewRing` function (its only caller).